File: batou-rules/testdata/fixtures/bench/lua/vulnerable/a03_command_injection.lua
-- Source: CWE-78 - OS Command Injection via os.execute in Lua
-- Expected: BATOU-LUA
-- OWASP: A03:2021 - Injection (Command Injection)

local input = arg[1]
local cmd = "ping -c 1 " .. input
os.execute(cmd)

File: batou-rules/testdata/fixtures/bench/cpp/vulnerable/a03_command_injection.cpp
// Source: CWE-78 - OS Command Injection via argv in C++
// Expected: BATOU-INJ, TAINT
// OWASP: A03:2021 - Injection (Command Injection)

#include <cstdlib>
#include <iostream>
#include <string>

void runUserCommand(int argc, char* argv[]) {
    if (argc < 2) return;
    std::string cmd = "ls -la ";
    cmd += argv[1];
    system(cmd.c_str());
}

int main(int argc, char* argv[]) {
    runUserCommand(argc, argv);
    return 0;
}
